### PR TITLE
Fetch dog breeds from server and display

### PR DESCRIPTION
**Pull request by qetiavanashvili**
1.  I added service property in viewmodel.
2. I removed fake data from array of DogBreed.
3. I created function getDogBreeds() in viewmodel and call DogBreedService's new function.

https://github.com/ashalva/ios.memenator/issues/2

### CODE REVIEW

**Review by ashalva — CHANGES_REQUESTED** [review]
@qetiavanashvili  your goal for this pr is to show real dogs, which are fetched from the server. Does it work? I am pretty sure, it does not.

**Comment by qetiavanashvili** [comment]
Haven't you checked the pr?

File: GeorgianMemenator/Network/DTOs/DogBreedsListDTO .swift [hunk 1]
```diff
@@ -11,7 +11,7 @@ struct DogBreedDTO: Codable {
         struct Weight: Codable {
             let max, min: Int
         }
-
+
```
Comment (ashalva) [inline GeorgianMemenator/Network/DTOs/DogBreedsListDTO .swift:14]: There should be 0 changes in this file, please try to revert the spaces.

File: GeorgianMemenator/Scenes/DogBreedsList/DogBreedsListView.swift [hunk 2]
```diff
@@ -19,13 +19,17 @@ struct DogBreedsListView: View {
...
+        })
+        .edgesIgnoringSafeArea(.bottom)
     }
 }
 
 struct DogBreedsListView_Previews: PreviewProvider {
     static var previews: some View {
-        DogBreedsListView(
-            viewModel: DogBreedsListViewModel(dogBreedsService: DogBreedsService())
+        MainView(
```
Comment (ashalva) [inline GeorgianMemenator/Scenes/DogBreedsList/DogBreedsListView.swift]: Why did you change it to `MainView`? it was correct

File: GeorgianMemenator/Scenes/DogBreedsList/DogBreedsListViewModel.swift [hunk 3]
```diff
@@ -11,50 +11,51 @@ import Foundation
...
-                            self?.isLoading = false
-                            self?.isImageLoading = false
-                        }
-                    }, receiveValue: { [weak self] val in
-                        self?.currentPost = val
+    private let dogBreedsService: DogBreedsServing
+    
+    @Published var dogBreeds = [DogBreed]()
+    
+    @Published private(set) var hasError: Bool = false
```
Comment (ashalva) [inline GeorgianMemenator/Scenes/DogBreedsList/DogBreedsListViewModel.swift]: This pr does not concentrate on:

1. Error
2. Loading
3. Image loading (it was needed in reddit vm)
4. Current post (it was needed in reddit vm)

Please remove these properties.

**Review by ashalva — COMMENTED** [review]
Good job, looks good 🚀  but there are some unnecessary changes which must be deleted.

Also, when you make some UI change, it would be nice to upload the image in pr description, so that I see how it looks when the project is built

File: GeorgianMemenator/Scenes/DogBreedsList/DogBreedsListView.swift (hunk 2 above)
Comment (qetiavanashvili) [reply GeorgianMemenator/Scenes/DogBreedsList/DogBreedsListView.swift]: I took it from reddit, i thought it was neeeded.

File: GeorgianMemenator/Scenes/DogBreedsList/DogBreedsListViewModel.swift (hunk 3 above)
Comment (qetiavanashvili) [reply GeorgianMemenator/Scenes/DogBreedsList/DogBreedsListViewModel.swift]: okey, I. will fix.

File: GeorgianMemenator/Scenes/DogBreedsList/DogBreedsListView.swift (hunk 2 above)
Comment (ashalva) [reply GeorgianMemenator/Scenes/DogBreedsList/DogBreedsListView.swift]: Yes, RedditView may have it incorrectly then, we can fix it afterwards. Not part of this pr 👍

File: GeorgianMemenator/Scenes/DogBreedsList/DogBreedsListViewModel.swift [hunk 4]
```diff
@@ -8,12 +8,45 @@
...
-        DogBreed(name: "nagazi",description: "a",minLife: 4, maxLife: 10,maleMinWeight: 12,maleMaxWeight: 14,femaleMinWeight: 23,femaleMaxWeight: 26),
-                 
-        DogBreed(name: "bulldog",description: "b",minLife: 4, maxLife: 10,maleMinWeight: 12,maleMaxWeight: 14,femaleMinWeight: 23,femaleMaxWeight: 26),
-        DogBreed(name: "labrador",description: "c",minLife: 4, maxLife: 10,maleMinWeight: 12,maleMaxWeight: 14,femaleMinWeight: 23,femaleMaxWeight: 26)
-    ]
+    private let dogBreedsService: DogBreedsServing
+    
+    @Published var dogBreeds = [DogBreed]()
+    
+//    @Published private(set) var hasError: Bool = false
```
Comment (ashalva) [inline GeorgianMemenator/Scenes/DogBreedsList/DogBreedsListViewModel.swift]: No need to comment these, just delete them

File: GeorgianMemenator/Scenes/DogBreedsList/DogBreedsListView.swift [hunk 5]
```diff
@@ -19,11 +19,17 @@ struct DogBreedsListView: View {
...
         }
+        .onAppear(perform: {
+            viewModel.initialFetch()
+        })
+        .edgesIgnoringSafeArea(.bottom)
     }
 }
 
 struct DogBreedsListView_Previews: PreviewProvider {
     static var previews: some View {
```
Comment (ashalva) [inline GeorgianMemenator/Scenes/DogBreedsList/DogBreedsListView.swift:30]: Please format the file by:

CMD + A
CTRL + I

File: GeorgianMemenator/Scenes/DogBreedsList/DogBreedsListViewModel.swift (hunk 4 above)
Comment (qetiavanashvili) [reply GeorgianMemenator/Scenes/DogBreedsList/DogBreedsListViewModel.swift]: I forgot to delete :(

File: GeorgianMemenator/Scenes/DogBreedsList/DogBreedsListView.swift (hunk 5 above)
Comment (qetiavanashvili) [reply GeorgianMemenator/Scenes/DogBreedsList/DogBreedsListView.swift:30]: okey

**Review by ashalva — APPROVED** [review]
